### ISSUE TITLE
Deleting characters is buggy in git-bash on Windows 10

### ISSUE THREAD

**Issue by jiangfengming** [platformio/platformio-atom-ide-terminal#108]
![2016-11-20_11-12-59](https://cloud.githubusercontent.com/assets/11750532/20460120/973ecf18-af14-11e6-9bc7-8344b89fe71a.gif)

This occurs when deleting too fast. Only in git-bash, powershell has no problem.

**Comment by jiangfengming** [comment]
It also effects Bash on Ubuntu on Windows.

**Comment by levsthings** [comment]
Having the same problems with git-bash. Deleting characters with backspace or traversing back and forth with arrow keys is very buggy.

**Comment by lancelot-lewis** [comment]
Having the same problem in win10

**Comment by Aetherall** [comment]
![2017-04-15_14-30-05](https://cloud.githubusercontent.com/assets/13298494/25063667/77e1a34e-21eb-11e7-9db0-ba8abcba2df7.gif)

Same problem over here !

Deleting characters with backspace or traversing back and forth with arrow keys is very buggy!

Where does it comes from?
Someone has an idea about how to fix it ?

Using git-bash on Windows 10

**Comment by ivalkenburg** [comment]
Same problem, also using git-bash

**Comment by drazik** [comment]
Yes, same here. git-bash on Windows 10 too.

**Comment by celestialorb** [comment]
Same problem here using cygwin on Windows 10.

**Comment by P403n1x87** [comment]
Same using git-bash on win10

**Comment by shayaShav** [comment]
Tried em all.. git-bash, cmder and Ubuntu bash on win 10. All buggy. Impossible to work with :(

**Comment by P403n1x87** [comment]
The steps at https://github.com/jeremyramin/terminal-plus/issues/15#issuecomment-144618245 seem to have "fixed" the problem in my case

**Comment by grig0ry** [comment]
Try disabling Cursor Blink option.

**Comment by clayms** [comment]
same problem.

Disabling Cursor Blink option does not fix.

Windows 10 Enterprise x64
Atom 1.25.0 x64
platformio-atom-ide-terminal 2.8.1
git version 2.14.3.windows.1

Shell Override - C:\Program Files\Git\bin\bash.exe

**Comment by clayms** [comment]
[Node.js docs](https://nodejs.org/api/os.html#os_os_eol) say that `os.EOL` is a "string constant defining the operating system-specific end-of-line marker:"
+ `\n` on POSIX
+ `\r\n` on Windows

**Comment by Onfire7** [comment]
Same issue when using cmd.exe as the shell...

Windows 10 X64
Atom 1.28.2
Platformio-ide-terminal 2.8.3

Enabling legacy mode on the command prompt fixes this issue...

**Comment by the-j0k3r** [comment]
@Onfire7 well 23 people had the issue fixed by this https://github.com/jeremyramin/terminal-plus/issues/15#issuecomment-144618245 which is what #575 proposes, yet the majority 90 people say this has no effect.

Its interesting results I think YMMV really

**Comment by P403n1x87** [comment]
It would be good to have a fix for this as the legacy more seems to break WSL. Perhaps the tech details [here](https://conemu.github.io/en/BashOnWindows.html) could be useful.

**Comment by the-j0k3r** [comment]
I agree but this project is no longer under active development, If someone submits a fix I dont mind pinging people to try get attention to it.

I wouldn't use WSL any and all malicious code bypasses all security if it comes via WSL.

**Comment by Onfire7** [comment]
@the-j0k3r Interesting... I would point out though, that people are FAR more likely to downvote something that isn't working, then to upvote something that is, so I would tend to believe that that leans more towards working for at least 50% of people. That said, 90+ is still a lot of people, so something else is definately going on...

I'm rather inclined to believe that those people that it didn't fix the issue for have 2+ seperate issues going on though... Mainly because it's known to cause the exact same issue with several other terminals... I guess the real question isn't so much "Will enabling legacy mode fix my terminal problems?" but rather "Is legacy mode required for my terminal to work on a fully up to date Windows 10 system, while using an alternate shell?" To answer that question:

### TL;DR:
Does anyone have a fully up to date Windows 10 system with legacy mode disabled, and an alternate shell setup in platformio-atom-ide-terminal that is working without the caret bugs? If so, what shell (with version, preferably) is it using?

**Comment by the-j0k3r** [comment]
>Does anyone have a fully up to date Windows 10 system with legacy mode disabled, and an alternate shell setup in platformio-atom-ide-terminal

Yes, me for instance, Ive tried using bash, cmd.com several times and cant duplicate this issue.

I cant duplicate 99% of issues reported, there are a couple of issues I have that are related to platformio-ide-terminal showing old text form a large interactive git rebase that clogs up the terminal view.

In any case, legacy console doesn't fix it, and like @P403n1x87 pointed out it breaks WSL (Windows subsystem for Linux)

I think people that have most of these issues I cant duplicate would do well in trying fresh Atom configurations like described in Atom debugging page, or start by disabling non core packages one by one testing in between and close reopen a terminal to see if issue persists, until the issue is gone, or you run out of external extension to disable,

TL;DR:
This project is no longer under active development (read #543 if you haven't already), and unless a more robust code is submitted by someone via PR, bugs that are caused by external packages or some Atom configuration or by some conflict with some external package, or because Atom doesn't fully support XYZ we are all SOL.

**Comment by P403n1x87** [comment]
I think at some point I have got something "working" by using the method described in the [ConEmu](https://conemu.github.io/en/BashOnWindows.html#connector) documentation:
~~~
set "PATH=%ConEmuBaseDirShort%\wsl;%PATH%" & %ConEmuBaseDirShort%\conemu-cyg-64.exe --wsl -cur_console:pm:/mnt
~~~
If it wasn't this then it must have been [wslbridge](https://github.com/rprichard/wslbridge). One way or the other I've got Bash from WSL working, but I couldn't use the arrows to scroll through the history, so not much use.
I would have a look at the code myself to see if there's anything that can be done, but at the minute I'm busy with other projects.

**Comment by ujeshurun** [comment]
Same issue on my pc. Windows 10, Git Bash. This is a great tool. Would love to see it work properly.

**Comment by marcusnewton** [comment]
Bump. Still happening with Git Bash on windows. If anyone knows of an alternate atom package for using Git Bash shells in atom hit me up

**Comment by the-j0k3r** [comment]
Please retest again with platformio-atom-ide-terminal 2.9.0.
These type issues all so far seem fixed with this release due to using node-pty-prebuilt instead unmaintained and outdated pty.js
